addons/web/static/src/legacy/js/core/cookie_utils.js

```diff
@@ -21,21 +21,40 @@ return {
         }
         return "";
     },
+    /**
+     * Check if cookie can be written.
+     *
+     * @param {String} type the type of the cookie
+     * @param {String} name the name of the cookie
+     * @returns {boolean}
+     */
+    checkCookie(type, name) {
+        return true;
+    },
     /**
      * Creates a cookie.
      *
      * @param {string} name the name of the cookie
      * @param {string} value the value stored in the cookie
      * @param {integer} ttl time to live of the cookie in millis. -1 to erase the cookie.
+     * @param {string} type the type of the cookies ('required' as default value)
      */
-    set_cookie(name, value, ttl) {
+    set_cookie(name, value, ttl = 31536000, type = 'required') {
+        if (isNaN(ttl)) {
+            console.error(`Unexpected TTL value ${ttl}`);
+            type = ttl;
```

Comment:
What is this line about ? It this in case someone wrongly call `set_cookie()` like `set_cookie('cook_name', 'cook_value', 'optional')` ?
If yes, seems overkill, if not could you explain?